### PR TITLE
Toolset update: VS 2019 16.10 Preview 1

### LINKED ISSUE #1637: <type_traits>: Remove _IS_LAYOUT_COMPATIBLE_SUPPORTED macro and related __builtin*() intrinsics
The original compiler intrinsics supporting [P0466R5](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p0466r5.pdf), were replaced by @JonCavesMSFT to better conform to the naming conventions in `<type_traits>`. The new names remove the `"__builtin"` prefix. To support the transition from the old names to the new ones, a compiler-provide macro is provided, `_IS_LAYOUT_COMPATIBLE_SUPPORTED`. #1575 used this macro to support both intrinsics names.

These compiler changes are expected to be included in Dev16.10 Preview 1. Once that version ships, the old names will be obsolete, and we should remove them as well and related `#ifdef _IS_LAYOUT_COMPATIBLE_SUPPORTED` in `<type_traits>`.

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* `stl/msbuild/stl_base/stl.files.settings.targets`: Our internal legacy build system can build `ppltasks.cpp` with `/permissive-` now.
  + The modern CMake build system doesn't need this change: https://github.com/microsoft/STL/blob/108a1127362cd14fd27f7df61aa23efba4f6a105/stl/CMakeLists.txt#L288-L290
* Update the Virtual Machine Scale Set.
  + `README.md`: Mention VS 2019 16.10 Preview 1 and Python 3.9.2 as dependencies.
  + `azure-devops/create-vmss.ps1`: Move back to `westus2` (after #1654) and try switching from `Standard_D32as_v4` to `Standard_D32ds_v4`. This should have comparable performance, and spot VMs are [approximately half the cost](https://azure.microsoft.com/en-us/pricing/details/virtual-machines/windows/).
  + `azure-devops/provision-image.ps1`: Update to PowerShell 7.1.2 and Python 3.9.2.
  + `azure-pipelines.yml`: Use the new pool.
* `stl/inc/yvals_core.h`: Increase required `_MSC_VER` to `1929`.
  + Note that VS 2019 16.8 and 16.9 shared `1928`, while 16.10 and 16.11 will share `1929`.
  + The internal "checked-in toolset" has already been updated.
* `stl/inc/xatomic.h`: Remove an `_mm_pause` declaration workaround.
* `stl/inc/type_traits`: Fix #1637 by removing the `_IS_LAYOUT_COMPATIBLE_SUPPORTED` checks; now we unconditionally use the new names.
* Remove workarounds for VSO-1271718, DevCom-1318357/VSO-1270432, DevCom-1318341/VSO-1269894, VSO-1275530, DevCom-1333853/VSO-1276453, DevCom-1336816/VSO-1277619, DevCom-1336819/VSO-1277640, and an unspecified bug that was fixed in 16.10 Preview 1.
* `tests/std/tests/P0218R1_filesystem/test.cpp`: Remove an outdated comment about "not a `PM_` constant"; that machinery was removed long ago.
* `tests/std/tests/Dev11_0836436_get_time/test.cpp`: Fix/add [locale.time.get.members] citations, noticed after merging #1620.
* `.gitattributes`: Mark `*.h` as C++ (they're being detected as C) and `*.cfg` as Python (they're being detected as HAProxy, whatever that is).
* **Update the `vcpkg` submodule for Boost 1.75.0.**
  + `CMakeLists.txt`: Require Boost 1.75.0.

### CODE REVIEW

**Review by CaseyCarter — APPROVED** [review]

File: stl/inc/type_traits [hunk 1]
```diff
@@ -1809,57 +1809,28 @@ inline constexpr bool is_nothrow_invocable_r_v =
 #ifndef __clang__ // TRANSITION, LLVM-48860
 // STRUCT TEMPLATE is_layout_compatible
 template <class _Ty1, class _Ty2>
-#ifdef _IS_LAYOUT_COMPATIBLE_SUPPORTED
```
[the remark this answers is not in the document]
Comment (MahmoudGSaleh) [reply stl/inc/type_traits:1812]: Good point. I think we should, since they won't be used anymore. @JonCavesMSFT.

**Review by MahmoudGSaleh — APPROVED** [review]
LGTM

File: stl/inc/type_traits (hunk 1 above)
[the remark this answers is not in the document]
Comment (StephanTLavavej) [reply stl/inc/type_traits:1812]: This is simple enough for me to do, I think.

**Comment by StephanTLavavej** [comment]
I reverted the `/headerUnit:angle` change, as it was causing the mysterious ICEs, and reverted those workaround additions (leaving only workaround removals in the test). I'll reduce and report the bug.

**Review by CaseyCarter — APPROVED** [review]

**Pull request merged**